### ISSUE TITLE
`Polygon` doesn't implement `PointDistance` (but `LineString` does)

### ISSUE THREAD

**Issue by lnicola** [georust/geo#1267]
I guess this explains why I never understood how to use `rstar`...

**Comment by urschrei** [comment]
This needs point-in-polygon. See https://github.com/georust/geo/pull/984#issuecomment-1432928362 for why this isn't easy: in short, a robust point-in-polygon implementation requires quite a lot of supporting code. We (obviously) have all of that _in `geo`_, and can't use it for `geo-types` because that would create a circular dependency. The alternative is copy-and-pasting everything that's required into `private_utils.rs`, which is not maintainable.

In short: https://github.com/georust/geo/pull/984#issuecomment-1747378040 is still the most sensible approach, except that it causes a lot of other problems.

**Comment by lnicola** [comment]
I guess a future rewrite of `geo` based on `geo-traits` will solve this?

That said, maybe `geo` should lose the `geo-types` dependency? I think it's used in a couple of different ways:

 - trait definitions (`CoordNum`)
 - complex types returned by some algorithms (`minimum_rotated_rect` returns `Polygon`)
 - "primitive" types used by some algorithms (`Line`, `Point`, `Triangle`)
 - tests
 - `PreparedGeometry` construction
 - anything else?

So should `geo` even use `Polygon` or expose `Polygon` in its implementation or public API?
